tests/integration/test_interface.py

```diff
@@ -6,77 +6,96 @@
+        (DATASETS.cps, generate_current_population_survey, True),
+        (DATASETS.cps, generate_current_population_survey, False),
+        (DATASETS.ssa, generate_social_security, True),
+        (DATASETS.ssa, generate_social_security, False),
+        (DATASETS.tax_w2_1099, generate_taxes_w2_and_1099, True),
+        (DATASETS.tax_w2_1099, generate_taxes_w2_and_1099, False),
+        (DATASETS.wic, generate_women_infants_and_children, True),
+        (DATASETS.wic, generate_women_infants_and_children, False),
+        ("DATASETS.tax_1040", "todo", True),
+        ("DATASETS.tax_1040", "todo", False),
     ],
 )
 def test_generate_dataset(
-    data_dir_name: str, noising_function: Callable, use_sample_data: bool, tmpdir
+    dataset: Dataset, noising_function: Callable, use_sample_data: bool, tmpdir
 ):
     """Tests that noised datasets are generated and as expected. The 'use_sample_data'
     parameter determines whether to use the sample data (if True) or
     a non-default root directory with multiple datasets to compile (if False)
     """
     if noising_function == "todo":
-        pytest.skip(reason=f"TODO: implement dataset {data_dir_name}")
+        pytest.skip(reason=f"TODO: implement dataset {dataset}")
 
-    sample_data_path = list(
-        (paths.SAMPLE_DATA_ROOT / data_dir_name).glob(f"{data_dir_name}*")
-    )[0]
+    sample_data_path = list((paths.SAMPLE_DATA_ROOT / dataset.name).glob(f"{dataset.name}*"))[
+        0
+    ]
 
     data = _load_data(sample_data_path)
 
     # Configure if default (sample data) is used or a different root directory
     if use_sample_data:
         source = None  # will default to using sample data
     else:
-        source = _generate_non_default_data_root(
-            data_dir_name, tmpdir, sample_data_path, data
-        )
+        source = _generate_non_default_data_root(dataset.name, tmpdir, sample_data_path, data)
 
-    noised_data = noising_function(seed=0, source=source)
-    noised_data_same_seed = noising_function(seed=0, source=source)
-    noised_data_different_seed = noising_function(seed=1, source=source)
+    noised_data = noising_function(seed=0, source=source, year=None)
+    noised_data_same_seed = noising_function(seed=0, source=source, year=None)
+    noised_data_different_seed = noising_function(seed=1, source=source, year=None)
 
     assert not data.equals(noised_data)
     assert noised_data.equals(noised_data_same_seed)
     assert not noised_data.equals(noised_data_different_seed)
 
-    # Check each column's dtype
-    for col in noised_data.columns:
-        expected_dtype = [c.dtype_name for c in COLUMNS if c.name == col][0]
+    # Check each column
+    shared_idx = pd.Index(set(data.index).intersection(set(noised_data.index)))
+    check_original = _reformat_dates_for_noising(data.loc[shared_idx], dataset)
+    check_noised = noised_data.loc[shared_idx]
+
+    for col_name in noised_data.columns:
+        col = [c for c in COLUMNS if c.name == col_name][0]
```

Comment:
Let's create a function in `__Columns` `get_column(name: str) -> Column:`

Reply:
Added the function to utilities - is that ok?

Reply:
Oh, nm, circular import. Will add method to the class